dexcom/fetch/runner.go

```diff
@@ -124,17 +124,16 @@ func (r *Runner) Run(ctx context.Context, tsk *task.Task) bool {
 	}
 
 	if !skipToAvoidDexcomBackup {
-		tsk.ClearError()
+		ResetTask(tsk)
 
 		if serverSessionToken, sErr := r.AuthClient().ServerSessionToken(); sErr != nil {
-			tsk.AppendError(errors.Wrap(sErr, "unable to get server session token"))
+			ErrorOrRetryTask(tsk, errors.Wrap(sErr, "unable to get server session token"))
```

Comment:
We shouldn't fail the task if we can't obtain a session token from our authentication service, otherwise we'll fail all tasks if the auth service is down for whatever reason. Retries should be unlimited too.